src/ra_type_traits.h

```diff
@@ -108,8 +127,38 @@ is_comparable_v{ detail::is_comparable<Comparable>::value };
 template<typename Comparable> _NODISCARD _CONSTANT_VAR
 is_nothrow_comparable_v{ detail::is_nothrow_comparable<Comparable>::value };
 
-} // namespace ra
+template<typename EqualityComparable, typename TestComparable> _NODISCARD _CONSTANT_VAR
+is_equality_comparable_with_v{ detail::is_equality_comparable_with<EqualityComparable, TestComparable>::value };
+
+template<typename EqualityComparable, typename TestComparable> _NODISCARD _CONSTANT_VAR
+is_nothrow_equality_comparable_with_v{ detail::is_nothrow_equality_comparable_with<EqualityComparable, TestComparable>::value };
+
+namespace detail {
 
+// Based off of the NullablePointer named requirement
+template<typename T, class = std::void_t<>>
+struct _NODISCARD is_nullable_pointer : std::false_type{};
+
+template<typename T>
+struct is_nullable_pointer<T, std::enable_if_t<
+    is_nothrow_equality_comparable<T>::value &&
+    std::is_nothrow_default_constructible_v<T> &&
+    std::is_nothrow_copy_constructible_v<T> &&
+    std::is_nothrow_copy_assignable_v<T> &&
+    std::is_nothrow_destructible_v<T> &&
+    std::is_nothrow_constructible_v<T, std::nullptr_t> &&
+    std::is_nothrow_assignable_v<T&, std::nullptr_t> &&
+    std::is_nothrow_constructible_v<T&&, std::nullptr_t> &&
+    std::is_same_v<decltype(std::declval<T&>() = nullptr), T&> &&
+    is_equality_comparable_with<T, std::nullptr_t>::value &&
+    is_equality_comparable_with<std::nullptr_t, T>::value
+>> : std::true_type{};
```

Comment:
I based this off of what  I read at cppreference. It's only used for SafeDelete now but will also be used for the custom handles later. It almost guarantees no exceptions will be thrown from a variable that satisfies this.